README.md

```diff
@@ -406,6 +406,8 @@ Getting metadata on any connection returns the following data structure:
 
 The following example illustrates how to use the `getMetadata` method.
 
+Please see the documentation for `Client.getMetadata` for a potential side-effect of the call.
```

Comment:
Can we reword this to something a little clearer. This seems a bit scary:

```
If you intend on passing a topic configuration to a topic you are explicitly fetching metadata for, be sure to created that topic prior to calling getMetadata on it. When fetching metadata for a topic, if a topic reference does not exist, one is created using the default config, which makes subsequent topic calls simply return that object.
```

We can and should also document it in the docblock as you have, but I think it's less cryptic if it is explicit in the README as well.